src/main/java/com/networknt/schema/JsonSchema.java

```diff
@@ -239,6 +239,20 @@ private boolean nodeContainsRef(String ref, JsonNode node) {
         return validators;
     }
 
+    /**
+     * A comparator that sorts validators, such such that 'properties' comes before 'required',
+     * so that we can apply default values before validating required.
+     */
+    private static Comparator<String> VALIDATOR_SORT = (lhs, rhs) -> {
```

Comment:
This change also changes the order of validators, so the order of validation messages will change.  If anyone was expecting the old order, they will break. Fortunately mvn test passes, and my code uses Matchers.containsInAnyOrder.

Reply:
I don't think the order of the messages is important.

Reply:
What about performance concerns?  Changing HashMap to TreeMap is changing O(1) to O(lg(N)).  Is there a performance test that can be run to test?

Reply:
The focus of this json-schema-validator is the small size of the JSON object and the number of the validators shouldn't that big. In most cases, we shouldn't worry about the performance. For the large-size JSON schemas, we might have an issue. I would suggest picking up one big JSON schema to test it with this [repo](https://github.com/networknt/json-schema-validator-perftest). If we see a performance issue, we can make it configurable to use either HashMap or TreeMap in the code. What do you think?

Reply:
(1) Using json-schema-validator-perftest as is which is version 1.0.49 time is around 430ms

(2) Upgrading to version 1.0.63 by changing the version to
```
            <version>1.0.63</version>
```
and the time is now around 465ms.

(3) Using my code by adding

            <scope>system</scope>
            <systemPath>/Users/snaran/main/Software/json-schema-validator/target/json-schema-validator-1.0.63.jar</systemPath>

now time is around 430ms.  Maybe the TreeMap makes things faster.  Hash map has to hash the whole string, which is O(N) of the whole string, whereas TreeMap only has to compare chars till it finds the first difference, so in theory TreeMap could be faster sometimes.  But who knows, just guessing.